measurements/api/measurements.py

```diff
@@ -143,26 +151,40 @@ def get_measurement(measurement_id):
             Autoclaved.filename.label('a_filename'),
             Autoclaved.autoclaved_no.label('a_autoclaved_no'),
 
-    ).filter(Measurement.id == measurement_id)\
+    ).filter(Measurement.msm_no == msm_no)\
         .join(Report, Report.report_no == Measurement.report_no)\
         .join(Autoclaved, Autoclaved.autoclaved_no == Report.autoclaved_no)
     try:
         msmt = q.one()
     except exc.MultipleResultsFound:
         current_app.logger.warning("Duplicate rows for measurement_id: %s" % measurement_id)
         msmt = q.first()
+    except exc.NoResultFound:
+        # XXX we should actually return a 404 here
+        raise BadRequest("No measurement found")
 
     r = requests.get(urljoin(current_app.config['AUTOCLAVED_BASE_URL'], msmt.a_filename),
             headers={"Range": "bytes={}-{}".format(msmt.frame_off, msmt.frame_off+msmt.frame_size)}, stream=True)
 
     # XXX use for streaming support lz4framed.Decompressor
     # @darkk how big can these lz4 frames even become? Should this be a concern?
     msmt_data = lz4framed.decompress(r.raw.read())[msmt.intra_off:msmt.intra_off+msmt.intra_size]
```

Comment:
For the record -- there is a bug in `lz4framed.decompress`, it does not raise exception on leftover data.

Reply:
#36